app/views/admin/_admin_options.html.erb

```diff
@@ -31,10 +31,33 @@
       </li>
     <% end %>
     <% if @work.present? %>
-      <li><%= link_to ts("Edit Tags and Language"), edit_tags_work_path(@work) %></li>
+      <li>
+        <%= link_to ts("Edit Tags and Language"), edit_tags_work_path(@work) %>
+      </li>
+      <% if @work.spam? %>
+        <li>
+          <%= link_to ts("Mark Not Spam"),
+                      { controller: "admin/user_creations",
+                        action: "set_spam",
+                        id: @work.id,
+                        creation_type: @work.class,
+                        spam: "false" } %>
```

Comment:
Since set_spam is now a put request, I think you need to add `method: :put` (not as part of the link hash, but after) to make these links work.